nav2_behaviors/include/nav2_behaviors/back_up.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIORS__BACK_UP_HPP_
+#define NAV2_BEHAVIORS__BACK_UP_HPP_
+
+#include <string>
+#include <memory>
+
+#include "nav2_tasks/back_up_task.hpp"
+#include "geometry_msgs/msg/twist.hpp"
+#include "std_msgs/msg/empty.hpp"
+#include "nav2_robot/robot.hpp"
+
+namespace nav2_behaviors
+{
+
+class BackUp : public rclcpp::Node
```

Comment:
Shouldn't there be an implementation class for all the recoveries to follow? Allows some structure or common code amongst recoveries as well as making meta-recovery nodes that have a configurable pointer vector of recoveries.

Reply:
Agree, I'll push a commit shortly that has a Behavior base clase. BackUp and Spin derive from it. I like the idea of meta-recoveries, and that's the idea behind identifying the minimum set of primitives from which we can compose more complex behaviors.